plugins/outputs/dynatrace/dynatrace.go

```diff
@@ -188,20 +192,31 @@ func (d *Dynatrace) Write(metrics []telegraf.Metric) error {
 				if err != nil {
 					continue
 				}
-				fmt.Fprintf(&buf, "%s", metricID)
-				// add the tag string
-				fmt.Fprintf(&buf, "%s", tagb.String())
-
-				// write measured value
+				// write metric id,tags and value
 				switch metric.Type() {
 				case telegraf.Counter:
-					fmt.Fprintf(&buf, " count,%v\n", value)
+					if lastvalue, ok := counts[metricID+tagb.String()]; ok {
+						// only send a counter if a lastvalue is found in the map
+						// if last value is found we can calc and send the delta value
+						if v, err := strconv.ParseFloat(lastvalue, 32); err == nil {
+							if v2, err := strconv.ParseFloat(value, 32); err == nil {
+								fmt.Fprintf(&buf, "%s%s count,delta=%f\n", metricID, tagb.String(), v2-v)
+							}
+						}
+					}
+					// put the current value into the map as last value
+					counts[metricID+tagb.String()] = value
 				default:
-					fmt.Fprintf(&buf, " %v\n", value)
+					fmt.Fprintf(&buf, "%s%s %v\n", metricID, tagb.String(), value)
 				}
 			}
 		}
 	}
+	sent++
+	// in typical interval of 10s, we will clean the counter state once in 24h which is 8640 iterations
+	if sent%8640 == 0 {
+		counts = make(map[string]string)
+	}
```

Comment:
you are right, but its just a periodic clear of the cache to not run into a mem leak issue if the number of metrics or dimensions increase in case of long running agent operation. So the interval does not really play a role here, I just chose it to not be too small.